core/am/logic/src/main/java/org/apache/syncope/core/logic/AuthModuleLogic.java

```diff
@@ -60,7 +59,8 @@ public AuthModuleTO update(final AuthModuleTO authModuleTO) {
         return binder.getAuthModuleTO(authModuleDAO.save(binder.update(authModule, authModuleTO)));
     }
 
-    @PreAuthorize("hasRole('" + AMEntitlement.AUTH_MODULE_LIST + "')")
+    @PreAuthorize("isAnonymous() or hasRole('" + AMEntitlement.AUTH_MODULE_LIST
```

Comment:
Hum I don't see any reason to apply such a change: what's the rationale, please?

Reply:
The PropertySourceLocator fails to call `list()` on AuthModuleService via the anonymous-user, and produces 403 error codes. Cross-checked with 1-2 other Logic components and they seem to follow a similar approach. Is there a better alternative?

Reply:
I see, then I'd remove `isAnonymous()` from the list, thus leaving the condition "`IdRepoEntitlement.ANONYMOUS` or `AMEntitlement.AUTH_MODULE_LIST`"